AccountingService/Controllers/UsersController.cs

```diff
@@ -15,130 +16,155 @@ public class UsersController : ControllerBase
-
+                var users = salaryConversionContext.User.ToArray(); // izvuci sve user iz tabele(user) 
+                return Ok(new
+                {
+                    Count = users.Length,
+                    Users = users,
+                });
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error: " + ex.Message);
+                return new StatusCodeResult(500);
+            }
         }
 
-
         [HttpGet("{id}")]
         public IActionResult GetUserById(int id)
         {
-            var user = salaryConversionContext.User.Where(u => u.ID == id).SingleOrDefault();
-            if (user == null)
+            try
             {
-                return NotFound();
-            }
+                var user = salaryConversionContext.User.Where(u => u.ID == id).SingleOrDefault();
 
-            return Ok(user);
 
+                return Ok(user);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error: " + ex.Message);
+                return new StatusCodeResult(500);
+            }
         }
 
-
-
-
         [HttpPost]
         public IActionResult CreateUser([FromBody] User request)
         {
-            salaryConversionContext.User.Add(request);
-            int changes = salaryConversionContext.SaveChanges();
-            if (changes == 0)
+            try
             {
-                return BadGateway();
+                salaryConversionContext.User.Add(request);
+                int changes = salaryConversionContext.SaveChanges();
+                return Ok();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error: " + ex.Message);
+                return new StatusCodeResult(500);
+
             }
-            return Ok();
         }
 
         [HttpPut]
         public IActionResult UpdateUser([FromBody] User request)
         {
-            salaryConversionContext.User.Update(request);
-            int changes = salaryConversionContext.SaveChanges();
-            if (changes == 0)
+            try
             {
-                return BadGateway();
+                salaryConversionContext.User.Update(request);
+                int changes = salaryConversionContext.SaveChanges();
+
+                return Ok();
             }
 
-            return Ok();
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error: " + ex.Message);
+                return new StatusCodeResult(500);
+            }
         }
 
         [HttpDelete("{id}")]
         public IActionResult DeleteUser(int id)
         {
-            var user = salaryConversionContext.User.Where(u => u.ID == id).SingleOrDefault();
-            if (user == null)
             {
-                return NotFound();
-            }
-            salaryConversionContext.User.Remove(user);
-            int changes = salaryConversionContext.SaveChanges();
-
-            if (changes == 0)
```

Comment:
Why did you remove this?

Reply:
done

Reply:
This is not done. Please re-add the check.